qa/setup_packages.py

```diff
@@ -314,11 +310,18 @@ def test_request(self, url):
         url = "://".join(url)
         request = Request(url)
         request.get_method = lambda : 'HEAD'
-        try:
-            _ = urlopen(request, timeout=60)
-            return url
-        except HTTPError:
-            return None
+        attempts = 3
+        while attempts:
+            try:
+                _ = urlopen(request, timeout=100)
+                return url
+            except HTTPError:
```

Comment:
so we allow URLErrors up to 3 attempts but we fail as soon as we get an HTTPError. Correct?

Reply:
Yes, URLErrors means timeout, HTTPError means that the destination doesn't exist.